### DISCUSSION TITLE
Leak API key on Realtime subscriptions

### DISCUSSION THREAD

**Discussion by BrunoFerreira95** [supabase/supabase#16715] [General]
When I use a real-time subscription on one page of next.

In the network with an F5, show the web socket? apikey=” apikey”

I make a post on LinkedIn, showing how to implement a role system login with supabase.

And a guy came to talk with me, and show this leak, I ready check in my app, and it’s real, it was showing the api key.

I make a quick search in the forum but I don’t find.

Guys warning. =/

**Comment by GaryAustin1** [comment]
The apikey shows up anytime used in a browser environment.  Which is why Supabase recommends using only the anon_key.
I assume your realtime code is running in a client on the browser.

**Reply by BrunoFerreira95** [reply]
That it, I am being walk for home, and think about this, today, I use an auth, with client side, but if I use the Realtime with serve side, fix the issue.
Thanks. soon I will try to solve these issues.
